src/main/java/net/mcreator/blockly/java/blocks/SetVariableBlock.java

```diff
@@ -35,25 +38,21 @@
 import java.util.stream.Collectors;
 
 public class SetVariableBlock implements IBlockGenerator {
+	private final String[] names;
 
-	@Override public void generateBlock(BlocklyToCode master, Element block) throws TemplateGeneratorException {
-		String type;
+	public SetVariableBlock() {
+		names = VariableElementTypeLoader.INSTANCE.getVariableTypes().stream().map(VariableElementType::getName)
+				.collect(Collectors.toList()).stream().map(s -> s = "variables_set_" + s).toArray(String[]::new);
+	}
 
-		String blocktype = block.getAttribute("type");
-		switch (blocktype) {
-		case "variables_set_number":
-			type = "NUMBER";
-			break;
-		case "variables_set_text":
-			type = "STRING";
-			break;
-		case "variables_set_logic":
-			type = "LOGIC";
-			break;
-		case "variables_set_itemstack":
-			type = "ITEMSTACK";
-			break;
-		default:
+	@Override public void generateBlock(BlocklyToCode master, Element block) throws TemplateGeneratorException {
+		String type = StringUtils.removeStart(block.getAttribute("type"), "variables_set_");
+		String javaType = new Dependency("",
+				VariableElementTypeLoader.INSTANCE.getVariableTypeFromString(type).getName())
+				.getType(master.getWorkspace());
+		if (javaType.equals("Object")) {

```

Comment:
If object is used, this is fine as one can always assign anything to an object. So I don't think this is 100% needed. Bigger problem is with get block where object will be used in eg. integer context. But this can't really happen so this check can probably be safely skipped.